### PR TITLE
Auth2.0 is done

### PR DESCRIPTION
**Pull request by idris-bohra**

### CODE REVIEW

File: src/components/endpoints/displayEndpoint.jsx [hunk 1]
```diff
@@ -848,6 +857,19 @@ class DisplayEndpoint extends Component {
...
+    if (!expirationTime) return false;
+    const generatedTime = new Date(generatedDateTime).getTime();
+    const expirationDateTime = generatedTime + expirationTime;
+    const currentTime = new Date().getTime();
+    const isExpired = currentTime > expirationDateTime;
+
+    return isExpired;
+  }
+
+  getFreshToken(singleTokenDetail) {
```
Comment (Prince8631) [inline src/components/endpoints/displayEndpoint.jsx:870]: where is it used ?

File: src/components/endpoints/displayEndpoint.jsx [hunk 2]
```diff
@@ -922,6 +944,43 @@ class DisplayEndpoint extends Component {
       url = this.replaceVariables(url, environment)
       url = this.addhttps(url)
       headers = this.replaceVariablesInJson(headers, environment)
+      // Start of Regeneration of AUTH2.0 Token
```
Comment (Prince8631) [inline src/components/endpoints/displayEndpoint.jsx:947]: make a function for it.

File: src/components/common/utility.js [hunk 3]
```diff
@@ -35,6 +35,61 @@ const orgKey = 'organisation'
 const orgListKey = 'organisationList'
 const proxyUrl = process.env.REACT_APP_PROXY_URL
 
+const untitledEndpointData = {
```
Comment (Prince8631) [inline src/components/common/utility.js:38]: why is it used in two places.

File: src/components/endpoints/displayEndpoint.jsx [hunk 4]
```diff
@@ -1525,9 +1594,15 @@ class DisplayEndpoint extends Component {
     if (title === 'content-type') {
       updatedHeaders[updatedHeaders.length - 1].value = this.identifyBodyType(value)
     }
-
     updatedHeaders.push(emptyHeader)
     const dummyData = this.props.endpointContent
+    if (dummyData?.authorizationData?.authorization?.oauth2) {
```
Comment (Prince8631) [inline src/components/endpoints/displayEndpoint.jsx:1599]: same is repeated on line 1546 make a function

File: src/components/common/utility.js (hunk 3 above)
Comment (idris-bohra) [reply src/components/common/utility.js:38]: for making default value of the untitledEndpointData

File: src/components/endpoints/displayEndpoint.jsx (hunk 4 above)
Comment (idris-bohra) [reply src/components/endpoints/displayEndpoint.jsx:1599]: old endpoint data has no key oauth2 inside the authorizationData so we make a check if oauth2 is not found then make an oauth2 inside the authorizationData object

File: src/components/endpoints/displayEndpoint.jsx (hunk 1 above)
Comment (idris-bohra) [reply src/components/endpoints/displayEndpoint.jsx:870]: okk will remove

File: src/components/endpoints/endpointApiService.js [hunk 5]
```diff
@@ -185,9 +80,82 @@ export async function setResponse(props, responseData) {
...
+
+  if (data.selectedGrantType === grantTypesEnums.passwordCredentials) {
+    body.grant_type = 'password'
+    body.username = data.username
+    body.password = data.password
+  }
+
+  try {
+    const { data: responseData } = await httpService.request({
+      url: `${process.env.REACT_APP_API_URL}/auth/token`,
```
[the remark this answers is not in the document]
Comment (idris-bohra) [reply src/components/endpoints/endpointApiService.js:129]: done

File: src/components/endpoints/displayEndpoint.jsx (hunk 2 above)
Comment (idris-bohra) [reply src/components/endpoints/displayEndpoint.jsx:947]: removed

File: src/components/common/utility.js (hunk 3 above)
Comment (idris-bohra) [reply src/components/common/utility.js:38]: removed

**Pull request closed without merging**